src/kafka-util/src/client.rs

```diff
@@ -72,6 +73,9 @@ impl ProducerContext for MzClientContext {
 pub struct BrokerRewritingClientContext<C> {
     inner: C,
     overrides: HashMap<BrokerAddr, BrokerAddr>,
+    /// SSH sessions associated with ssh overrides.
+    // TODO(guswynn) and close() as well
+    ssh_session_tokens: Vec<Sender<()>>,
```

Comment:
This makes for a fuzzy abstraction boundary! Can we move in either the more specific or more general direction?

1. Change this to `tokens: Box<dyn Any>`, and extend the API with an `add_token` method.
2. Create a `SshTunnelingClientContext` that wraps a `BrokerRewritingClientContext`, with a `new` method that takes the tunnel parameters and an `add_broker` method that takes each broker address and establishes a tunnel to that broker (and adds a rewrite to the wrapped broker rewriting context).

(2) seems theoretically nicer, but I suspect (1) is a lot less typing, because I think there are number of places in the Kafka sink/source implementation where we need to name a concrete Kafka context type.

Reply:
I think ill go with 1, just for the ease, I suspect that we may want other kinds of tokens someday